google/cloud/testing_util/integration_test.cc

```diff
@@ -0,0 +1,30 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/testing_util/integration_test.h"
+#include "google/cloud/log.h"
+
+namespace google {
+namespace cloud {
+inline namespace GOOGLE_CLOUD_CPP_NS {
+namespace testing_util {
+
+void IntegrationTest::TearDown() {
+  if (HasFailure()) ::google::cloud::LogSink::Instance().Flush();
```

Comment:
This is part of the reason I wasn't overly keen on the idea ... this log dump may be long after the failing expectation of interest, so the relevant traces may have disappeared, or at least won't be the most recent ones.  But, that shouldn't stop us from doing what we can.

Reply:
I tried to connect an event listener, but had no luck getting it to work. This looks promising:

https://github.com/google/googletest/blob/b7d472f1225c5a64943821d8483fecb469d3f382/googletest/include/gtest/gtest.h#L1126

Reply:
Ack.  I failed to even find such an interface, so, yes, promising.  I'll see if I can learn anything.  Thanks,